http_test.go

```diff
@@ -156,3 +157,18 @@ func TestRedirectNotWhenHTTPS(t *testing.T) {
 
 	assert.Equal(t, http.StatusOK, res.StatusCode, "status code should be %d, got: %d", http.StatusOK, res.StatusCode)
 }
+
+func TestGracefulShutdown(t *testing.T) {
+	opts := NewOptions()
+	stop := make(chan struct{}, 1)
+	srv := Server{Handler: http.DefaultServeMux, Opts: opts, stop: stop}
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		srv.ServeHTTP()
+	}()
+
+	stop <- struct{}{} // emulate catching signals
+	wg.Wait()
```

Comment:
In case this functionality is broken, this test case would hang right? Do you think we can add a timeout somehow?

Reply:
To find broken case, I added assert to check chanel length and add an idiomatic for sync.WaitGroup with timeout.